### PR TITLE
[Merged by Bors] - feat: Add lemmas about `coeff_single_X_pow`

### PR DESCRIPTION
**Pull request by BoltonBailey**
Adds some lemmas about the coefficient of polynomials of the form `X s ^ n`.

Upstreamed from another project.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/MvPolynomial/Basic.lean [hunk 1]
```diff
@@ -738,6 +738,20 @@ theorem coeff_X_mul (m) (s : σ) (p : MvPolynomial σ R) :
...
+  simp only [coeff_X_pow, single_eq_single_iff]
+
+@[simp]
+lemma coeff_single_X [DecidableEq σ] (s s' : σ) (n : ℕ) :
+    (X s).coeff (R := R) (Finsupp.single s' n) = if n = 1 ∧ s = s' then 1 else 0 := by
+  rw [coeff_X']
+  simp only [single_eq_single_iff]
+  congr 1
+  simp only [one_ne_zero, false_and, or_false, eq_iff_iff]
+  tauto
```
Comment (Ruben-VandeVelde) [inline Mathlib/Algebra/MvPolynomial/Basic.lean:753]:
```suggestion
  simpa [eq_comm, and_comm] using coeff_single_X_pow s s' 1 n
```

**Review by Ruben-VandeVelde — APPROVED** [review]
lgtm

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**